xapian-applications/omega/omega.cc

```diff
@@ -174,7 +175,11 @@ try {
 
     hits_per_page = 0;
     auto val = cgi_params.find("HITSPERPAGE");
-    if (val != cgi_params.end()) hits_per_page = atol(val->second.c_str());
+    if (val != cgi_params.end()) {
+	if (!parse_unsigned(val->second.c_str(),hits_per_page)) {
+	    throw "HITSPERPAGE parameter must be >= 0\n";
```

Comment:
I don't think we want a newline of these exception messages (here and several other places).  None of the existing cases seem to include a newline, and one is added when the exception message is printed, so yours result in an extra newline, e.g.:

```
olly@gemse:~/git/xapian/xapian-applications/omega$ ./omega HITSPERPAGE=-1
Content-Type: text/html

Exception: HITSPERPAGE parameter must be &gt;= 0

olly@gemse:~/git/xapian/xapian-applications/omega$
```

(The `&gt;` is expected - the exception message is escaped for HTML to avoid potential cross-site scripting attacks through error messages which include parameter values).

Reply:
thanks i will fix that.